cot-cli/src/utils.rs

```diff
@@ -153,30 +168,38 @@ impl WorkspaceManager {
 
     fn parse_workspace(cargo_toml_path: &Path, manifest: Manifest) -> WorkspaceManager {
         assert!(manifest.workspace.is_some());
-        let workspace = manifest.workspace.as_ref().unwrap();
+        let workspace = manifest
+            .workspace
+            .as_ref()
+            .expect("workspace is known to be present");
+
+        let workspace_root = cargo_toml_path
+            .parent()
+            .expect("Cargo.toml should always have a parent");
         let package_manifests = workspace
             .members
             .iter()
             .map(|member| {
-                let member_path = cargo_toml_path
-                    .parent()
-                    .expect("Cargo.toml should always have a parent")
+                let member_path = workspace_root
                     .join(member)
-                    .join("Cargo.toml");
+                    .canonicalize()
```

Comment:
I would suggest not canonicalizing paths if possible. People tend to symlink stuff inside workspaces for whatever reasons, and it can be seen even in some popular crates - see https://github.com/serde-rs/serde/blob/master/serde_derive_internals/src for instance.

Canonicalizing paths will break `parent()`/`join()` and may produce incorrect paths.